wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/environment/server/compose/deserializer/CommandDeserializer.java

```diff
@@ -0,0 +1,107 @@
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.api.environment.server.compose.deserializer;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.TreeNode;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.TextNode;
+
+import org.eclipse.che.api.environment.server.compose.ComposeServiceImpl;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.lang.String.format;
+import static java.util.Arrays.asList;
+
+/**
+ * Custom deserializer for parsing {@link ComposeServiceImpl#command}
+ * to list command words in the yaml compose file.
+ * <p>Command can be:</p>
+ * <ul>
+ * <li>string array:</li>
+ *
+ * <pre>
+ *  image: codenvy/mysql
+ *  environment:
+ *   MYSQL_USER: petclinic
+ *   MYSQL_PASSWORD: password
+ *  mem_limit: 2147483648
+ *  command: [service, mysql, start]
+ * </pre>
+ *
+ * <li>simple string line:</li>
+ *
+ * <pre>
+ *  image: codenvy/mysql
+ *  environment:
+ *   MYSQL_USER: petclinic
+ *   MYSQL_PASSWORD: password
+ *  mem_limit: 2147483648
+ *  command: service mysql start
+ * </pre>
+ * </ul>
+ *
+ *
+ * See more for parsing compose file
+ * {@link org.eclipse.che.api.environment.server.compose.ComposeFileParser}.
+ * <p>Note: this deserializer works for json too.</p>
+ *
+ * @author Alexander Andrienko
+ */
+public class CommandDeserializer extends JsonDeserializer<List<String>> {
+
+    private static final String SPLIT_COMMAND_REGEX = "[ \n\r]+";
+
+    /**
+     * Parse command field from the compose yaml file to list command words.
+     *
+     * @param jsonParser
+     *         json parser
+     * @param ctxt
+     *         deserialization context
+     * @throws IOException
+     *         in case I/O error. For example element to parsing contains
+     *         invalid yaml field type defined for this field by yaml document model.
+     * @throws JsonProcessingException
+     */
+    @Override
+    public List<String> deserialize(JsonParser jsonParser, DeserializationContext ctxt) throws IOException {
+        TreeNode tree = jsonParser.readValueAsTree();
+
+        if (tree.isArray()) {
+            return toCommand((ArrayNode)tree);
+        }
+        if (tree instanceof TextNode) {
+            TextNode textNode = (TextNode)tree;
+            return asList(textNode.asText().trim().split(SPLIT_COMMAND_REGEX));
+        }
+        throw new RuntimeException(format("Field '%s' should be simple text or string array.", jsonParser.getCurrentName()));
```

Comment:
Deserializer already declares exceptions that can be thrown. Runtime exception can be replaced with something else I suppose. Does Jackson have an example of deserializators creation to find out what is the proper way to throw an error.